packages/lib-webpack/src/webpack/DynamicRemotePlugin.ts

```diff
@@ -93,63 +119,77 @@ export class DynamicRemotePlugin implements WebpackPluginInstance {
 
   private readonly sharedModules: WebpackSharedObject;
 
-  private readonly entryCallbackSettings: Required<PluginEntryCallbackSettings>;
+  private readonly moduleFederationLibraryType: string;
+
+  private readonly entryCallbackSettings: PluginEntryCallbackSettings;
+
+  private readonly entryScriptFilename: string;
+
+  private readonly pluginManifestFilename: string;
 
   constructor(options: DynamicRemotePluginOptions = {}) {
     const adaptedOptions: Required<DynamicRemotePluginOptions> = {
-      pluginMetadata: options.pluginMetadata ?? 'plugin.json',
-      extensions: options.extensions ?? 'extensions.json',
+      pluginMetadata: options.pluginMetadata ?? 'plugin-metadata.json',
+      extensions: options.extensions ?? 'plugin-extensions.json',
       sharedModules: options.sharedModules ?? {},
+      moduleFederationLibraryType: options.moduleFederationLibraryType ?? 'jsonp',
       entryCallbackSettings: options.entryCallbackSettings ?? {},
+      entryScriptFilename: options.entryScriptFilename ?? 'plugin-entry.js',
```

Comment:
@vojtechszocs NIT - we may want to consider putting the default options in an object and then spreading the provided options over it to make this a bit easier to read. Not blocking.

Reply:
Agreed. One note to add here, the updated code should retain the type safety, i.e. there must be a default value for each declared option within the "defaults" object. We can do this as a follow-up.